__tests__/api/BundleContent/Bundle.test.ts

```diff
@@ -130,30 +135,693 @@ describe("Bundle01", () => {
-        const bund = new Bundle("__tests__/__resources__/ExampleBundle03");
+        const bund = new Bundle("__tests__/__resources__/ExampleBundle03", true, true);
 
         // Add a definition
         bund.addNodejsappDefinition("NodeName", "__tests__/__resources__/ExampleBundle03/Artefact1", 1000);
 
         // Check the manifest as JSON
-        expect(JSON.stringify(bund.getManifest())).toMatch("{\"manifest\":{\"xmlns\":\"http://www.ibm.com/xmlns/prod/cics/bundle\",\"$t\":\"\",\"define\":[{\"name\":\"NodeName\",\"type\":\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\",\"path\":\"nodejsapps/NodeName.nodejsapp\"}]}}");
+        expect(JSON.stringify(bund.getManifest())).toMatchSnapshot();
         // Check the manifest as XML
-        expect(bund.getManifestXML()).toMatch("<manifest xmlns=\"http://www.ibm.com/xmlns/prod/cics/bundle\"><define name=\"NodeName\" type=\"http://www.ibm.com/xmlns/prod/cics/bundle/NODEJSAPP\" path=\"nodejsapps/NodeName.nodejsapp\"></define></manifest>");
+        expect(bund.getManifestXML()).toMatchSnapshot();
+    });
+    it("should warn that an existing bundle cant be overwritten", () => {
+
+        // Create a Bundle
+        const bund = new Bundle("__tests__/__resources__/ExampleBundle01", false, false);
+
+        let err: Error;
+        try {
+          bund.prepareForSave();
+        }
+        catch (error) {
+          err = error;
+        }
+
+        // Check the output as JSON
+        expect(err.message).toMatchSnapshot();
+    });
+    it("should be able to save new definitions", () => {
+
+        // Create a Bundle
+        const bund = new Bundle("__tests__/__resources__/ExampleBundle02", true, true);
+
+        // Add a definition
+        bund.addDefinition({name: "name1", type: "type1", path: "__tests__/__resources__/ExampleBundle02/Artefact1"});
+        bund.prepareForSave();
+
+        // Check the output as JSON
+        expect(JSON.stringify(bund.getManifest())).toMatchSnapshot();
+    });
+});
+
+
+// Note, the following tests mock the file-system. Snapshot based tests are unlikely to
+// work as the jest implementation will itself need to interact with the filesystem.
+describe("MockedFilesystemTests", () => {
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+    it("should tolerate META-INF directory not existing", () => {
+
+        // Mocks for the manifest - META-INF exists
+        const existsSpy = jest.spyOn(fs, "existsSync").mockImplementationOnce(() => ( false ));
+        // Mocks for the manifest - Bundle dir writable
+        const accessSpy = jest.spyOn(fs, "accessSync").mockImplementationOnce(() => ( true ));
+
+        let err: Error;
+        try {
+          const bund = new Bundle("__tests__/__resources__/ExampleBundle01", false, false);
+          bund.prepareForSave();
+        }
+        catch (error) {
+          err = error;
+        }
+
+        expect(err).toBeUndefined();
+        expect(existsSpy).toHaveBeenCalledTimes(1);
```

Comment:
I think these sort of checks could lead to a brittle test... suggest a `toHaveBeenCalledWith(path)` would be more useful. Then we verify the implementation is doing what we want without constraining it to the single call.

Reply:
See above, I found those tests to be essential. Yes it's closely coupled to the code... but that's just the nature of unit tests.